Assets/Scripts/HttpTopicClient.cs

```diff
@@ -0,0 +1,136 @@
+        this.authToken = authToken;
+    }
+
+    public void Publish(string cacheName, string topicName, string message)
+    {   
+        var contentType = "text/plain";
+        if (message.Contains("<sprite"))
+        {
+            Debug.Log("sending sprite message as binary");
+            contentType = "application/octet-stream";
+        }
+        var request = UnityWebRequest.Post(
+            "https://" + endpoint + "/topics/" + cacheName + "/" + topicName,
+            message,
+            contentType
+        );
+        request.SetRequestHeader("Authorization", authToken);
+        request.SendWebRequest();
+    }
+
+    public HttpTopicSubscription Subscribe(string cacheName, string topicName, Action<string> onMessage, Action<string> onError)
+    {
+        var sub = new HttpTopicSubscription();
+        sub.Poller = Poll(cacheName, topicName, onMessage, onError, sub);
+        return sub;
+    }
+
+    private IEnumerator Poll(
+        string cacheName, string topicName, Action<string> messageCallback, Action<string> errorCallback, HttpTopicSubscription parent
+    )
+    {
+        var sequenceNumber = 1;
+        var baseUri = "https://" + endpoint + "/topics/" + cacheName + "/" + topicName + "?sequence_number=";
+        while (true) {
+            var uri = baseUri + sequenceNumber;
+            Debug.Log("Polling " + uri);
+            UnityWebRequest www = UnityWebRequest.Get(uri);
+            www.SetRequestHeader("Authorization", authToken);
+            yield return www.SendWebRequest();
+
+            if (parent.Cancelled)
+            {
+                Debug.Log("subscription to " + cacheName + "/" + topicName + " has been cancelled.");
+                yield break;
+            }
+            if (www.result != UnityWebRequest.Result.Success)
+            {
+                errorCallback(www.error);
+            }
+            else
+            {
+                var itemJson = www.downloadHandler.text;
+                var stuff = JObject.Parse(itemJson);
+                foreach (var item in stuff["items"])
+                {
+                    if (item["discontinuity"] != null)
+                    {
+                        sequenceNumber = (int)item["discontinuity"]["new_topic_sequence"];
+                    }
+                    else
+                    {
+                        sequenceNumber = (int)item["item"]["topic_sequence_number"] + 1;
+                        var jsonDict = item["item"]["value"].ToObject<Dictionary<string, object>>();
+                        if (jsonDict.ContainsKey("text"))
+                        {
+                            messageCallback(jsonDict["text"].ToString());
+                        }
+                        else if (jsonDict.ContainsKey("binary"))
+                        {
+                            var obj = item["item"]["value"]["binary"].ToObject<byte[]>();
+                            var result = System.Text.Encoding.UTF8.GetString(obj);
```

Comment:
This definitely won't work with real binary data.

Reply:
This is working just fine for messages originating within the demo and from the SDKs.